### PR TITLE
[Docs][MPS] Add mps environment variable table

### PR DESCRIPTION
**Pull request by qqaatw**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #129105
* #129007
* __->__ #129008

### CODE REVIEW

File: docs/source/mps_environment_variables.rst [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  * - Variable
+    - Description
+  * - ``PYTORCH_DEBUG_MPS_ALLOCATOR``
+    - If set to ``1``,set allocator logging level to verbose.
+  * - ``PYTORCH_MPS_HIGH_WATERMARK_RATIO``
+    - High watermark ratio for MPS allocator. By default, it is set to 1.7.
+  * - ``PYTORCH_MPS_LOW_WATERMARK_RATIO``
+    - Low watermark ratio for MPS allocator. By default, it is set to 1.4 if the memory is unified and set to 1.0 if the memory is discrete.
+  * - ``PYTORCH_MPS_FAST_MATH``
+    - If set to ``1``, enable fast math for MPS metal kernels. See section 1.6.3 in https://developer.apple.com/metal/Metal-Shading-Language-Specification.pdf for precision implications.
```
Comment (malfet) [inline docs/source/mps_environment_variables.rst:20]: Hmm, this is not aligned with what you've defined in previous API, where setting it to zero would also enable fast-math, wouldn't it?

**Review by malfet — APPROVED** [review]
Looks good to me

File: docs/source/mps_environment_variables.rst (hunk 1 above)
Comment (qqaatw) [reply docs/source/mps_environment_variables.rst:20]: Ah let me rectify it!

**Comment by hvaara** [comment]
Thanks a lot for this! Really helpful 😃

**Pull request closed without merging**